esp/services/ws_access/ws_accessService.cpp

```diff
@@ -2162,6 +2162,436 @@ bool Cws_accessEx::onResourcePermissions(IEspContext &context, IEspResourcePermi
+
+        resp.setSuccess(true);
+        resp.setResult("Successfully added a view.");
+    }
+    catch (IException* e)
+    {
+        StringBuffer eMsg;
+        e->errorMessage(eMsg);
+        e->Release();
+        resp.setSuccess(false);
+        resp.setResult(eMsg);
+        return false;
+    }
+        
+    return true;
+}
+
+bool Cws_accessEx::onDeleteView(IEspContext &context, IEspDeleteViewRequest &req, IEspDeleteViewResponse &resp)
+{
+    try
+    {
+        CLdapSecManager* secmgr = queryLDAPSecurityManager(context);
+        
+        if(secmgr == NULL)
+            throw MakeStringException(ECLWATCH_INVALID_SEC_MANAGER, MSG_SEC_MANAGER_IS_NULL);
+
+        checkUser(context);
+
+        secmgr->deleteView(req.getViewname());
+
+        resp.setSuccess(true);
+        resp.setResult("Successfully deleted a view.");
+    }
+    catch (IException* e)
+    {
+        StringBuffer eMsg;
+        e->errorMessage(eMsg);
+        e->Release();
+        resp.setSuccess(false);
+        resp.setResult(eMsg);
+        return false;
+    }
+        
+    return true;
+}
+
+bool Cws_accessEx::onQueryViewColumns(IEspContext &context, IEspQueryViewColumnsRequest &req, IEspQueryViewColumnsResponse &resp)
+{
+    try
+    {
+        CLdapSecManager* secmgr = queryLDAPSecurityManager(context);
+        
+        if(secmgr == NULL)
+            throw MakeStringException(ECLWATCH_INVALID_SEC_MANAGER, MSG_SEC_MANAGER_IS_NULL);
+
+        checkUser(context);
+        
+        IArrayOf<IEspViewColumn> viewColumns;
+        StringArray files, columns;
+
+        secmgr->queryViewColumns(req.getViewname(), files, columns);
+
+        ForEachItemIn(i, files)
+        {
+            Owned<IEspViewColumn> oneViewColumn = createViewColumn();
+            oneViewColumn->setViewname(req.getViewname());
+            oneViewColumn->setFilename(files.item(i));
+            oneViewColumn->setColumnname(columns.item(i));
+            viewColumns.append(*oneViewColumn.getLink());
+        }
+
+        resp.setSuccess(true);
+        resp.setResult("Successfully queried all view columns.");
+        resp.setViewcolumns(viewColumns);
+    }
+    catch (IException* e)
+    {
+        StringBuffer eMsg;
+        e->errorMessage(eMsg);
+        e->Release();
+        resp.setSuccess(false);
+        resp.setResult(eMsg);
+        return false;
+    }
+
+    return true;
+}
+
+bool Cws_accessEx::onAddViewColumn(IEspContext &context, IEspAddViewColumnRequest &req, IEspAddViewColumnResponse &resp)
+{
+    try
+    {
+        CLdapSecManager* secmgr = queryLDAPSecurityManager(context);
+        
+        if(secmgr == NULL)
+            throw MakeStringException(ECLWATCH_INVALID_SEC_MANAGER, MSG_SEC_MANAGER_IS_NULL);
+
+        checkUser(context);
+
+        const char* origFileName = req.getFilename();
+        const char* origColumnName = req.getColumnname();
+
+        int fileLength = strlen(origFileName);
```

Comment:
picky: Should use size_t for the length of a string - otherwise you are likely to get warnings in more modern compilers.